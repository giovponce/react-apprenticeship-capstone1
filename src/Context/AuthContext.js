import React, { useState, useContext, useEffect, useCallback, createContext } from "react";
import { storage } from '../utils/storage';
import { AUTH_STORAGE_KEY } from '../utils/constants';


export const AuthContext = createContext('');

function useAuth() {
    return useContext(AuthContext);
}

// eslint-disable-next-line react/prop-types
export function AuthProvider({ children }){
    const [authenticated, setAuthenticated] = useState(false);

    useEffect(() => {
        const lastAuthenticated = storage.get(AUTH_STORAGE_KEY);
        const isAuthenticated = Boolean(lastAuthenticated);

        setAuthenticated(isAuthenticated);
    }, []);

    const login = useCallback(() => {
        setAuthenticated(true);
        storage.set(AUTH_STORAGE_KEY, true);
    }, []);

    const logout = useCallback(() => {
        setAuthenticated(false);
        localStorage.clear();
        storage.set(AUTH_STORAGE_KEY, false);
        window.location.reload();
    }, []);

    return (
        <AuthContext.Provider value={{login, logout, authenticated}}>
            {children}
        </AuthContext.Provider>
    )
}

export { useAuth };


