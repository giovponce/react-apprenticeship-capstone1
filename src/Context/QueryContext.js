import { createContext, useState, useContext } from "react";

export const QueryContext = createContext();
const QueryUpdateContext = createContext();

export function useQuery() {
  return useContext(QueryContext);
}

export function useQueryUpdate() {
  return useContext(QueryUpdateContext);
}

export function QueryProvider({children}) {
  const [query, setQuery] = useState('Wizeline');

  function changeQuery(newQuery) {
    setQuery(newQuery);
  }


  return (
  <QueryContext.Provider value={query}>
    <QueryUpdateContext.Provider value={changeQuery}>
      {children}
    </QueryUpdateContext.Provider>
  </QueryContext.Provider>
  );
}
