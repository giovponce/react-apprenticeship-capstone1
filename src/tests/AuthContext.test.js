import { render } from '@testing-library/react';
import { AuthContext } from '../Context/AuthContext';


describe('user context', () => {
    test('initial authenticated should be false', () => {
        const { container } = render(<AuthContext.Provider value={{ state: { authenticated: false } }}>
            <div>
                <h1>Hello</h1>
            </div>
        </AuthContext.Provider>);
        expect(container.firstChild).toMatchSnapshot(); //eslint-disable-line
    });


})