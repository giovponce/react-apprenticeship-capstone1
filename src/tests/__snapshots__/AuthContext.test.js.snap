// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user context initial authenticated should be false 1`] = `
<div>
  <h1>
    Hello
  </h1>
</div>
`;
